site/src/Librecores/ProjectRepoBundle/Controller/ProjectController.php

```diff
@@ -221,6 +221,20 @@ public function viewAction(Request $request, $parentName, $projectName)
             'isHostedOnGithub' => GithubRepoCrawler::isGithubRepoUrl($p->getSourceRepo()->getUrl()),
         ];
 
+        // Retrieve the project and set the code quality metrics
```

Comment:
I didn't really follow the full discussion, but why is there a database update in the view controller? That looks wrong.

Reply:
My suggestion is to move it to GitRepoCrawler, we can persist it there itself.